src/MainWindow.vala

```diff
@@ -903,14 +906,15 @@ namespace Terminal {
 
         private void action_open_in_browser () requires (current_terminal != null) {
             var uri = get_current_selection_link_or_pwd ();
-            string? to_open = Utils.sanitize_path (uri, current_terminal.get_shell_location ());
-            if (to_open != null) {
+            var to_open = Utils.sanitize_path (uri, current_terminal.get_shell_location (), true);
+            AppInfo.launch_default_for_uri_async.begin (to_open, null, null, (obj, res) => {
```

Comment:
I think you need to add the default launch context for this to send focus to the browser propertly
```suggestion
            var context = Gdk.Display.get_default ().get_app_launch_context ();
            AppInfo.launch_default_for_uri_async.begin (to_open, context, null, (obj, res) => {
```